contracts/MechMarketplace.sol

```diff
@@ -196,6 +206,29 @@ contract MechMarketplace is IErrorsMarketplace {
         );
     }
 
+    /// @dev Calculates payment for request delivery.
+    /// @param mech Delivery mech address.
+    /// @param payment Payment amount.
+    function _calculatePayment(
+        address mech,
+        uint256 payment
+    ) internal virtual returns (uint256 mechPayment, uint256 marketplaceFee) {
```

Comment:
In the function _calculatePayment we should also add the marketplaceFee to collectedFee, no ? 
Otherwise drain() does nothing it seems.

Reply:
Good catch!